src/com/hms_networks/americas/sc/localdatafiles/LocalDataFileManager.java

```diff
@@ -0,0 +1,476 @@
+    String day = String.valueOf(calendar.get(Calendar.DAY_OF_MONTH));
+    String hour = String.valueOf(calendar.get(Calendar.HOUR));
+
+    // Pad month, day and hour with leading zero if only one digit
+    month = month.length() == 1 ? "0" + month : month;
+    day = day.length() == 1 ? "0" + day : day;
+    hour = hour.length() == 1 ? "0" + hour : hour;
+
+    return localDataFileRootFolder
+        + "/"
+        + year
+        + "/"
+        + month
+        + "/"
+        + day
+        + "/"
+        + LocalDataFileConstants.LOCAL_DATA_FILE_NAME_PREFIX
+        + year
+        + month
+        + day
+        + hour
+        + LocalDataFileConstants.LOCAL_DATA_FILE_EXTENSION;
+  }
+
+  /**
+   * Stops local data file management and returns when the local data file management thread has
+   * terminated.
+   *
+   * @throws InterruptedException if unable to wait for local data file management thread to
+   *     terminates
+   */
+  public void stopLocalDataFileManagement() throws InterruptedException {
+    doProcessing = false;
+    join();
+  }
+
+  /**
+   * Deletes outdated local data files from the file system. The configured value of {@link
+   * #retainNumberMonths} is subtracted from the current month to get the first month which will be
+   * deleted.
+   */
+  private void deleteOutdatedFiles() {
+    // Get current date/time
+    Calendar calendar = Calendar.getInstance();
+    long localTimeMillis =
+        System.currentTimeMillis() - LocalTimeOffsetCalculator.getLocalTimeOffsetMilliseconds();
+    calendar.setTimeInMillis(localTimeMillis);
+
+    // Get date sections (add 1 to month - based on 0)
+    int year = calendar.get(Calendar.YEAR);
+    int month = calendar.get(Calendar.MONTH) + 1;
+
+    /*
+     * Subtract AZConnectorConsts.LOCAL_DATA_FILE_DELETE_AFTER_MONTHS from current date to get month to begin deleting.
+     */
+    for (int x = 0; x <= retainNumberMonths; x++) {
+      month--;
+      if (month == 0) {
+        month = 12;
+        year--;
+      }
+    }
+
+    /*
+     * Delete up to AZConnectorConsts.LOCAL_DATA_FILE_DELETE_NUM_MONTHS months of outdated data
+     */
+    List yearFolderPathsToCleanup = new ArrayList();
+    for (int x = 0; x <= LocalDataFileConstants.LOCAL_DATA_FILE_MAX_MONTHS_DELETE; x++) {
+      // Get month as string and pad with leading zero if only one digit
+      String monthString = String.valueOf(month);
+      String paddedMonth = monthString.length() == 1 ? "0" + monthString : monthString;
+
+      // Get folder name to delete
+      String folderToDeletePath = localDataFileRootFolder + "/" + year + "/" + paddedMonth;
+
+      // Get file object
+      File folderToDelete = new File(folderToDeletePath);
+
+      // Delete folder contents and folder
+      FileManager.recursivelyDeleteFileFolder(folderToDelete);
+
+      // Subtract month to get next to delete
+      month--;
+      if (month == 0) {
+        month = 12;
```

Comment:
Is there no constant you can use for this?

Reply:
constant added